aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -35,7 +35,26 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
 
   const bool is_self_scalar = self.dim() == 0;
   const bool is_other_scalar = other.dim() == 0;
-
+  bool needsScatter = false;
```

Comment:
Why is this needed only for BinaryOps?

Reply:
Can't we derive this information from the Alias info .